docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -548,201 +850,148 @@ func setGreeting(accessibleState PrecompileAccessibleState, caller common.Addres
- We can search (`CTRL F`) for `ADD YOUR PRECOMPILE HERE`.
-
-This file helps set up stateful precompiles that can be activated as part of a network upgrade and
-related helper functions.
-
-Let's create our precompile key and name it `helloWorldKey`. Precompile keys are used to reference
-each of the possible stateful precompile types. The below code snippet can be copied and pasted to
-overwrite the default code.
+In [`precompile/contract/utils.go`](https://github.com/ava-labs/subnet-evm/blob/helloworld-official-tutorial-v2/precompile/contract/utils.go#L19-L20)
+we have `WriteGasCostPerSlot` and `ReadGasCostPerSlot`.
 
 ```go
+// Gas costs for stateful precompiles
 const (
-	contractDeployerAllowListKey precompileKey = iota + 1
-	contractNativeMinterKey
-	txAllowListKey
-	feeManagerKey
-	helloWorldKey
-	// ADD YOUR PRECOMPILE HERE
-	// {yourPrecompile}Key
+	WriteGasCostPerSlot = 20_000
+	ReadGasCostPerSlot  = 5_000
 )
 ```
 
-We should also add our precompile key to our `precompileKeys` slice. We use this slice to iterate
-over the keys and activate the precompiles. The below code snippet can be copied and pasted to
-overwrite the default code.
+`WriteGasCostPerSlot` is the cost of one write such as modifying a state storage slot.
 
-<!-- markdownlint-disable MD013 -->
+`ReadGasCostPerSlot` is the cost of reading a state storage slot.
 
-```go
-// ADD YOUR PRECOMPILE HERE
-var precompileKeys = []precompileKey{contractDeployerAllowListKey, contractNativeMinterKey, txAllowListKey, feeManagerKey, helloWorldKey /* {yourPrecompile}Key */}
-```
-
-<!-- markdownlint-enable MD013 -->
+This should be in your gas cost estimations based on how many times the precompile function does a
+read or a write. For example, if the precompile modifies the state slot of its precompile address
+twice then the gas cost for that function would be `40_000`.
 
-We should add our precompile to the `PrecompileUpgrade` struct. The `PrecompileUpgrade` is a helper
-struct embedded in `UpgradeConfig` and represents each of the possible stateful precompile types
-that can be activated or deactivated as part of a network upgrade. The below code snippet can be
-copied and pasted to overwrite the default code.
+On top of these gas costs, we also have to account for the gas costs of AllowList gas costs. These
+are the gas costs of reading and writing permissions for addresses in AllowList. These are defined
+under [`precompile/allowlist/allowlist.go`](https://github.com/ava-labs/subnet-evm/blob/helloworld-official-tutorial-v2/precompile/allowlist/allowlist.go#L28-L29).
+By default these are added to the default gas costs to the state-change functions (SetGreeting)
+of the precompile. Meaning that these functions will cost an additional `ReadAllowListGasCost` in order
+to read permissions from the storage. If you don't plan to read permissions from the storage then
+you can omit these.
 
```

Comment:
Should we talk about the dangers of not accurately putting the correct gas costs?

Reply:
good idea, added